packages/web/src/common/store/pages/track/sagas.ts

```diff
@@ -44,77 +40,36 @@ function* getRestOfLineup(permalink: string, ownerHandle: string) {
   )
 }
 
-function* watchFetchTrack() {
+function* watchFetchTrackSucceeded() {
   yield* takeEvery(
-    trackPageActions.FETCH_TRACK,
-    function* (action: ReturnType<typeof trackPageActions.fetchTrack>) {
-      const {
-        trackId,
-        handle,
-        slug,
-        canBeUnlisted,
-        forceRetrieveFromSource,
-        withRemixes = true
-      } = action
-      try {
-        let track
-        if (!trackId) {
-          if (!(handle && slug)) return
-          track = yield* call(retrieveTrackByHandleAndSlug, {
-            handle,
-            slug,
-            withStems: true,
-            withRemixes,
-            withRemixParents: true,
-            forceRetrieveFromSource
-          })
-        } else {
-          const ids = canBeUnlisted
-            ? [{ id: trackId, url_title: slug, handle }]
-            : [trackId]
-
-          const tracks: Track[] = yield* call(retrieveTracks, {
-            trackIds: ids,
-            canBeUnlisted,
-            withStems: true,
-            withRemixes,
-            withRemixParents: true
-          })
-          track = tracks && tracks.length === 1 ? tracks[0] : null
-        }
-        const isReachable = yield* select(getIsReachable)
-        if (!track) {
-          if (isReachable) {
-            yield* put(pushRoute(NOT_FOUND_PAGE))
-          }
-        } else {
-          yield* put(trackPageActions.setTrackId(track.track_id))
-          // Add hero track to lineup early so that we can play it ASAP
-          // (instead of waiting for the entire lineup to load)
-          yield* call(addTrackToLineup, track)
-          if (isReachable) {
-            yield* fork(
-              getRestOfLineup,
-              track.permalink,
-              handle || track.permalink.split('/')?.[1]
-            )
-          }
-          yield* put(trackPageActions.fetchTrackSucceeded(track.track_id))
-        }
-      } catch (e) {
-        console.error(e)
-        yield* put(
-          trackPageActions.fetchTrackFailed(trackId ?? `/${handle}/${slug}`)
-        )
-      }
+    trackPageActions.FETCH_TRACK_SUCCEEDED,
+    function* (action: trackPageActions.FetchTrackSucceededAction) {
```

Comment:
no need to change it, but fwiw this logic looks very doable from a useEffect

Reply:
addTrackToLineup and getRestOfLineup are saga fns though, im a bit scared rn

Reply:
yeah not for today, but it's maybe not so bad idk